app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt

```diff
@@ -126,6 +127,10 @@ class EventsViewModel(
                     }
                 }
             }, {
+                if ((it as HttpException).code() == 401) {
```

Comment:
As said by codacy, this is an Unsafe cast. You can't know if the exception is HttpException, it can be IOException IllegalArgumentException, anything and you just casting it is waiting for a crash to happen.

Also, it's the developer's responsibility to handle any error except 500, where you can show Unfortunately, something went wrong, because you actually don't know what happened. 

All other errors must be handled. As a user, what am I supposed to do with a toast that says `HTTP 401 UNAUTHORIZED`?

I should either see `You must log in to see this {item}`, or if I am already logged in, then `Your session has expired, please login again`

And maybe being redirected to login screen

Reply:
> I should either see `You must log in to see this {item}`, or if I am already logged in, then `Your session has expired, please login again`

I was not sure it is because of the user's login state. It shows when I try to change API and do not clear the app data. And events should load if the user is not logged in. I think it is due to the wrong Authorization which needs to re-request the Authenticator.

Reply:
> As said by codacy, this is an Unsafe cast. You can't know if the exception is HttpException, it can be IOException IllegalArgumentException, anything and you just casting it is waiting for a crash to happen.

Sorry about that.

Reply:
> I try to change API and do not clear the app data.

Then clear the data when API change happens

Reply:
Mean clear the app data programmatically?

Reply:
Can we re-request the authenticator and reload events?

Reply:
No, because access token from one API won't work in another. This is very easily handled in Orga App. We simply show login screen anytime we encounter 401. Reason may be expired token or changing API base URL. Fixes every case

Reply:
Yes, It shows in case of login state only. For that error, we can log out the user and reload events. Because events should also load in not logged in state. :+1:

Reply:
Will fetch setting and test :+1:
Should I fetch setting at the starting of the app and store into the database or fetch every time when required?

Reply:
Why fetching settings?

Reply:
Sorry wrong PR.

Reply:
Updated